apps/dashboard/src/data/projects/actions.js

```diff
@@ -21,13 +22,21 @@ export function updateProject( projectId, project ) {
 
 export function* saveAndUpdateProject( projectId, project ) {
 	try {
+		yield fetchOngoing();
 		const response = yield saveProject( projectId, project );
 		const id = projectId || response.data.id;
 
-		return updateProject( id, {
+		yield updateProject( id, {
 			...project,
 			id,
 		} );
+
+		if ( ! projectId ) {
+			// this should only happen when creating a new project
+			yield redirect( `/${ id }` );
```

Comment:
Like I mentioned in our last conversation, the redirect doesn't really belong in the state. I mean, it would work, but it'd be better to do it from the component since here you don't have the context where you want to redirect to or why.

While it works, if we ever needed to dispatch this action from anywhere else than the editor for whatever reason this will become a problem. Hence in the original version it was part of the `Editor` component.

Reply:
Not entirely convinced about the route not being part of the state. It is something that actually *changes* and it makes for different behaviors. If you think of it, it's very similar to `lastUpdatedItemId`: defines what you're working against and is something on the *editor* (`ui`) state. I'm ok sticking with `lastUpdatedItemId`, but a router state that states what route (project) I'm working on makes more sense to me.
Thinking on switching from editor to results: the route changes and we'll still need some selector to know where are we and what we're working against.